hibernate-jcache/src/main/java/org/hibernate/cache/jcache/internal/JCacheRegionFactory.java

```diff
@@ -182,7 +199,14 @@ protected boolean isStarted() {
 	}
 
 	@Override
-	protected void prepareForUse(SessionFactoryOptions settings, Map<String,Object> configValues) {
+	protected void prepareForUse(SessionFactoryOptions settings, Map<String, Object> configValues) {
+		final ClassLoader serviceClassLoader = getServiceClassLoader( settings );
+		if ( serviceClassLoader != null ) {
+			preInitializedCacheManager = JCacheHelper.locateStandardCacheManager();
+			oldJsr107CacheClassLoader = Caching.getDefaultClassLoader();
+			Caching.setDefaultClassLoader( serviceClassLoader );
+		}
```

Comment:
IMO, the `JCacheHelper` class should rather be deprecated and/or moved to the test sources than being used explicitly here.


```suggestion
		final ClassLoader serviceClassLoader = getServiceClassLoader( settings );
		oldJsr107CacheClassLoader = Caching.getDefaultClassLoader();
		Caching.setDefaultClassLoader( serviceClassLoader );
```